source/utilities.cc

```diff
@@ -768,8 +769,9 @@ namespace aspect
               else
                 {
                   grid_spacing = new_temp_coord - temp_coord;
-                  if (std::abs(grid_spacing - first_grid_spacing) >
-                      std::numeric_limits<double>::epsilon()*std::abs(grid_spacing+first_grid_spacing)*2)
+                  // Compare current grid spacing with first grid spacing,
+                  // taking into account roundoff of the read-in coordinates
+                  if (grid_spacing / first_grid_spacing < 0.995 || grid_spacing / first_grid_spacing > 1.005)
```

Comment:
I think the previous version of this check was more intuitive. Is your current one not equivalent to replacing `std::numeric_limits<double>::epsilon()` by 0.005? I would be favouring that if you need a larger tolerance.